polkadot/runtime/common/src/paras_registrar/mod.rs

```diff
@@ -665,6 +677,21 @@ impl<T: Config> Pallet<T> {
 	}
 }
 
+impl<T: Config> OnNewHead for Pallet<T> {
+	fn on_new_head(id: ParaId, _head: &HeadData) -> Weight {
+		// mark the parachain locked if the locked value is not already set
+		let mut writes = 0;
+		if let Some(mut info) = Paras::<T>::get(id) {
+			if info.locked.is_none() {
+				info.locked = Some(true);
+				Paras::<T>::insert(id, info);
+				writes += 1;
+			}
+		}
+		T::DbWeight::get().reads_writes(1, writes)
```

Comment:
Nit: RFC claims this does not have any performance impact, yet we add some weight on each and every included candidate.

Reply:
IMO RFC should only specify the big O notation complexity. In this case, it is the same, a small constant change is implementation detail.